scripts/module/formapplications/sequencer-database-viewer.js

```diff
@@ -1,179 +1,227 @@
-                return `<mark>${str}</mark>`
-            }) : entry);
-        });
+        return el;
+      }
+    );
+  }
+
+  static show(inFocus = false) {
+    if (!game.user.isTrusted) return;
+    for (const app of Object.values(ui.windows)) {
+      if (app instanceof this) {
+        return app.render(true, { focus: inFocus });
+      }
     }
+    return new this().render(true);
+  }
 
-    /* -------------------------------------------- */
-
-    /** @override */
-    getData() {
-        let data = super.getData();
-        data.packs = this.packs;
-        data.entries = this.getFilteredEntries();
-        return data;
+  static get isVisible() {
+    for (const app of Object.values(ui.windows)) {
+      if (app instanceof this) return app;
     }
-
-    /** @override */
-    activateListeners(html) {
-        super.activateListeners(html);
-
-        let filter = html.find('select[name="pack-select"]');
-        let input = html.find('input[name="search-field"]');
-        this.player = html.find('.database-player');
-        this.image = html.find('.database-image');
-
-        let filterDebounce = debounce(() => {
-            this.search = input.val();
-            this.filter = filter.val();
-            this.refreshView(html);
-        }, 500)
-
-        filter.change(filterDebounce);
-        input.keyup(function (e) {
-            filterDebounce()
-        });
-
-        let self = this;
-        html.find('.btn_play').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.playAsset.bind(self)(entry);
-        });
-        html.find('.btn_copy_databasepath').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.copyText($(this), entry, false);
-        });
-        html.find('.btn_copy_filepath').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.copyText($(this), entry, true);
-        });
+  }
+
+  /** @override */
+  static get defaultOptions() {
+    return foundry.utils.mergeObject(super.defaultOptions, {
+      title: "Sequencer Database Viewer",
+      template: `modules/sequencer/templates/sequencer-database-template.html`,
+      classes: ["dialog"],
+      width: 900,
+      height: 425,
+    });
+  }
+
+  getSearchRegex() {
+    return new RegExp(this.search, "gu");
+  }
+
+  getFilteredEntries() {
+    const searchParts = this.search.split("|");
+    const { search, filter } = this;
+    return this.entries.filter((part) => {
+      return (
+        (filter === "all" || filter === part.pack) &&
+        (search === "" ||
+          part.entry.match(this.getSearchRegex())?.length >= searchParts.length)
+      );
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  getData() {
+    const data = super.getData();
+    data.packs = this.packs;
+    return data;

```

Comment:
Guide: entries is removed here because we're not using it anymore. Entries are manually managed